### ISSUE TITLE
Not showing events on October of any year.

### ISSUE THREAD

**Issue by jorgecasariego** [Applandeo/Material-Calendar-View#334]
I'm experiencing the same problem as #236 and #76 

Events are shown in all months except October

I'm using this version of the library:
`com.applandeo:material-calendar-view:1.9.0-alpha03`

The ViewModel in the following code is returning a List of type Calendar
```
      viewModel.getCalendarDays().observe(viewLifecycleOwner, Observer {
            calendarView.selectedDates = it
            calendarView.requestLayout()
        })
```

An this is the result in October
![image](https://user-images.githubusercontent.com/2581647/96787070-87a3a300-13c7-11eb-9121-16cc083cf418.png)

And the correct one in November
![image](https://user-images.githubusercontent.com/2581647/96787099-938f6500-13c7-11eb-8434-f8e4a31b4526.png)

**Comment by Afur** [comment]
@jorgecasariego Could you provide us code that will be repetitive on our side? Right now I'm not able to reproduce this issue on my side so my work on the fix is very limited.

**Comment by jorgecasariego** [comment]
@Afur I changed the sample code to add a few days in October.

**[many_days_picker_activity.xml](https://github.com/Applandeo/Material-Calendar-View/blob/master/sample/src/main/res/layout/many_days_picker_activity.xml)**

```
<?xml version="1.0" encoding="utf-8"?>
<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    tools:context="com.applandeo.materialcalendarsampleapp.MainActivity">

    <com.applandeo.materialcalendarview.CalendarView
        android:id="@+id/calendarView"
        android:layout_width="0dp"
        android:layout_height="wrap_content"
        android:layout_marginTop="8dp"
        app:abbreviationsBarColor="#303F9F"
        app:abbreviationsLabelsColor="@android:color/white"
        app:daysLabelsColor="@android:color/white"
        app:headerColor="#1A237E"
        app:layout_constraintLeft_toLeftOf="parent"
        app:layout_constraintRight_toRightOf="parent"
        app:layout_constraintTop_toTopOf="parent"
        app:pagesColor="#3F51B5"
        app:eventsEnabled="false"
        app:selectionColor="@android:color/white"
        app:selectionLabelColor="#3F51B5"
        app:anotherMonthsDaysLabelsColor="#3F51B5"
        app:todayLabelColor="#AEEA00"
        app:type="many_days_picker" />

    <Button
        android:id="@+id/getDateButton"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:text="Get date"
        app:layout_constraintLeft_toLeftOf="parent"
        app:layout_constraintRight_toRightOf="parent"
        app:layout_constraintTop_toBottomOf="@id/calendarView" />

</androidx.constraintlayout.widget.ConstraintLayout>
```

**[ManyDaysPickerActivity.java](https://github.com/Applandeo/Material-Calendar-View/blob/master/sample/src/main/java/com/applandeo/materialcalendarsampleapp/ManyDaysPickerActivity.java)**

```
     // This is the new code. The rest is the same as the repository
    private List<Calendar> getSelectedDays() {
        List<Calendar> calendars = new ArrayList<>();

        for (int i = 0; i < 20; i++) {
            Calendar calendar = DateUtils.getCalendar();
            calendar.set(Calendar.MONTH, 9);
            calendar.set(Calendar.DAY_OF_MONTH, i);
            calendars.add(calendar);
        }

        return calendars;
    }
```

**Result**

Of the 19 days it only shows the first 4 days selected. The other months are working without problem.

![image](https://user-images.githubusercontent.com/2581647/101193074-c9687000-363a-11eb-9a67-7197783011f5.png)

**Comment by pavlomelnyk** [comment]
Hi @jorgecasariego . I tested samples that you provided and did not reproduced. Also made more general method form your example to test few years and all the dates were displayed correctly. 
`    private List<Calendar> getSelectedDays() {
        List<Calendar> calendars = new ArrayList<>();

        for (int y = 0; y < 3; y++) {
            for (int m = 0; m < 12; m++) {
                for (int i = 1; i < 29; i++) {
                    Calendar calendar = DateUtils.getCalendar();
                    calendar.set(Calendar.MONTH, m);
                    calendar.set(Calendar.YEAR, 2018+y);
                    calendar.set(Calendar.DAY_OF_MONTH, i);
                    calendars.add(calendar);
                }
            }
        }

        return calendars;
    }`

Also tried set my local date on October 2020 to simulate time when you spot this issue and also was not able to reproduce. 
Could you please try to reproduce this bug on a clear sample code of this library. Maybe it is some modification that you apply that is causing this issue. Please provide more details.

**Comment by jorgecasariego** [comment]
This is super weird. I just cloned the whole project and all I did was change the code in getSelectedDays() and this is the result

![image](https://user-images.githubusercontent.com/2581647/101834509-3ab39180-3b19-11eb-9762-9ed09d3e2d6d.png)

<img src="https://user-images.githubusercontent.com/2581647/101834679-6f274d80-3b19-11eb-82d5-f10da2ad846c.gif" width="480" height="854" />

Tested in:
- Pixel XL API 29 (Emulator) x86_64

**Comment by jorgecasariego** [comment]
Hi @pavlomelnyk , 

I found something related to TimeZone. 

If I change for example to the region **United States** and open the app again the calendar works without problem. 

![image](https://user-images.githubusercontent.com/2581647/103558247-23a35d80-4e93-11eb-8e93-4e965ed00fec.png)
![image](https://user-images.githubusercontent.com/2581647/103558296-31f17980-4e93-11eb-954d-95d4b49e3502.png)


The problem I think is specifically with the country I use: Paraguay

![image](https://user-images.githubusercontent.com/2581647/103558324-403f9580-4e93-11eb-8d3a-af18dfdc97a9.png)

**Comment by kormateusz** [comment]
Hi @jorgecasariego please check new 1.9.0-rc02 version.

**Comment by kormateusz** [comment]
Fixed in 1.9.0-rc03

**Comment by jorgecasariego** [comment]
thanks @kormateusz
